app/uk/gov/hmrc/exports/models/declaration/submissions/Action.scala

```diff
@@ -38,10 +90,50 @@ object Action {
   implicit val readLocalDateTimeFromString: Reads[ZonedDateTime] = implicitly[Reads[LocalDateTime]]
     .map(ZonedDateTime.of(_, ZoneId.of("UTC")))
 
-  implicit val writes = Json.writes[Action]
+  private val allWrites = (JsPath \ "id").write[String] and
```

Comment:
Curious as to why you need customs read/writes?

Reply:
To read/write from an record to any one of the actions based on requestType

Reply:
The theory is that rather than having one action that has all different rules based on its requestType, we can read the requestType and better model an action accordingly